pennylane_lightning/src/simulator/Measures.hpp

```diff
@@ -157,6 +160,35 @@ class Measures {
         return std::real(expected_value);
     };
 
+#ifdef _ENABLE_KOKKOS
+    /**
+     * @brief Expected value of a Sparse Hamiltonian.
+     *
+     * @param row_map row_map array.
+     *                The j element encodes the number of non-zeros above row j.
+     * @param entries column indices of the non-zero elements.
+     * @param values  non-zero elements.
+     * @return fp_t
+     */
+    fp_t expval(const std::vector<Util::index_type> &row_map,
+                const std::vector<Util::index_type> &entries,
+                const std::vector<CFP_t> &values) {
+        if (original_statevector.getLength() != (row_map.size() - 1)) {
+            throw std::invalid_argument(
+                "Statevector and Hamiltonian have incompatible sizes.");
+        }
+        auto operator_vector = Util::apply_Sparse_Matrix(
+            original_statevector.getData(), original_statevector.getLength(),
+            row_map.data(), row_map.size(), entries.data(), values.data(),
+            values.size());
+
+        CFP_t expected_value = Util::innerProdC(
+            original_statevector.getData(), operator_vector.data(),
+            original_statevector.getLength());
+        return std::real(expected_value);
```

Comment:
Do other parts of the codebase return real or complex here? Might be worth checking and doing the same as they